.github/workflows/app-test-coverage-pr.yml

```
@@ -0,0 +1,87 @@
+# /********************************************************************************
+# * Copyright (c) 2023 Contributors to the Eclipse Foundation
+# *
+# * See the NOTICE file(s) distributed with this work for additional
+# * information regarding copyright ownership.
+# *
+# * This program and the accompanying materials are made available under the
+# * terms of the Apache License, Version 2.0 which is available at
+# * https://www.apache.org/licenses/LICENSE-2.0.
+# *
+# * Unless required by applicable law or agreed to in writing, software
+# * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# * License for the specific language governing permissions and limitations
+# * under the License.
+# *
+# * SPDX-License-Identifier: Apache-2.0
+# ********************************************************************************/
+
+name: Application test and coverage -- PR context
+
+on:
+  pull_request:
+    paths:
+      - 'src/**'
+
+jobs:
+  test-app:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set up Java 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+
+      - name: Test app
+        env:
+          GITHUB_USERNAME: ${{ vars.GITHUB_ACTOR }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: ./gradlew test -PtestIgnoreFailures > test_output.txt
```

Just generally, I don't see any particular value in letting the pipeline run if the build or test stages fail. What is the motivation behind this? What is the value of the test report if it is failing?

---

In general, the test result are shown in the log of the run.
But the result of the code coverage is not shown there, and you have to get there to see at least the test results.
Here, the test results (tests and coverage) are nicely presented in the run (report and as summary) and partly in the PR.

BUT, when the tests fail, no results are generated... So, there is nothing to show.
Everyone would have to visit the run logs to see the test results there (but still no code coverage results)

---

That would be the https://github.com/marketplace/actions/junit-report-action I suggested in the other comment. It feels weird doing all the extra work, just to get to the report which is already generated by junit.

---

Your suggested action says: "This action processes JUnit XML test reports ...", but I don't ignore the test failure, I **won't get any files** this action can process (in this configuration)... 
Therefore I need to either separate the gradle tasks "test" and "jacocoTestReport" (what I did first, but you complained) or skip the execution stop on test failure.
In the case of test failure, I have to check, whether at least one test fails afterwards and then let the pipeline fail itself.

If there is a more elegant solution, please let me know.

---

On the contrary. JUnit test reports are always created. You can verify this locally by adding a `Assertion.assertTrue(false)` to any test and running the tests `./gradlew clean test`. Then you'll see all the `.xml` files of all run tests in the output directory `build/test-results/test/TEST*.xml`. These files are always produced except when you provide the quick-fail attribute. You can upload them with the suggested action to get a better insight into the failure.
Then we keep the test report as previously discussed.

In the case of a failure:
1. run tests
2. fail
3. upload junit report
4. stop execution (no jacocoReport)

in the case of success:
1. run the tests
2. succeed
3. upload junit report
4. continue execution (produce jacocoReport)

---

We had a sync with @pmoscode and talked about the whole structure of the PR. We've found that there was some miscommunication regarding the goal and the expected results.

At the end we've decided to split the test execution and the jacoco report in gradle, add the jacoco report to the task execution of tests in Taskfile and revert the pipeline to an earlier version, where the tests would run -> generate the junit reports -> generate the binary for jacoco, then run another task to generate the html and xml results for jacoco. At that point a failure in test would cause the whole pipeline to fail and we'd still have the junit test report and the coverage reports as part of the PR.